backupccl: elide expensive ShowCreate call in SHOW BACKUP

adityamaru opened the pull request:
In https://github.com/cockroachdb/cockroach/issues/88376 we see the call to `ShowCreate` taking ~all the time on a cluster with
2.5K empty tables. In all cases except `SHOW BACKUP SCHEMAS` we do not
need to construct the SQL representation of the table's schema. This
results in a marked improvement in the performance of `SHOW BACKUP`
as can be seen in https://github.com/cockroachdb/cockroach/issues/88376#issuecomment-1254164321.

Fixes: https://github.com/cockroachdb/cockroach/issues/88376

Release note (performance improvement): `SHOW BACKUP` on a backup containing
several table descriptors is now more performant

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/88293)
<!-- Reviewable:end -->

adityamaru committed:
backupccl: elide expensive `ShowCreate` call in `SHOW BACKUP`

In #88376 we see this call taking ~all the time on a cluster with
2.5K empty tables. In all cases except `SHOW BACKUP SCHEMAS` we do not
need to construct the SQL representation of the table's schema. This
results in a marked improvement in the performance of `SHOW BACKUP`
as can be seen in https://github.com/cockroachdb/cockroach/issues/88376#issuecomment-1254164321.

Fixes: #88376

Release note (performance improvement): `SHOW BACKUP` on a backup containing
several table descriptors is now more performant

adityamaru commented:
TFTR!

bors r=stevendanna

postamar commented:
Did we really want to merge all these `tracing.ChildSpan` calls in this PR? These hardly seem desirable.

postamar commented:
Thanks for this explanation! I was assuming that they were left in by mistake. I'm surprised you can't use reflection to auto-generate the span name, wouldn't that be preferable?
